### DISCUSSION TITLE
Kube + Code Server

### DISCUSSION THREAD

**Discussion by ozzyst** [coder/code-server#4501] [Q&A] [answered]
Hey there!

I use Kubernetes for deployment Code Server, and I can't use env. I try to set `"env": {...}` into the launch.json, but it gives nothing, and my code can't access vars from env. I have read about we can't change vars inside a pod, so is there a problem?

Any idea how can I use it?

**Comment by ozzyst** [comment]
But I can set var from the terminal inside the pod... I'm confused. Any idea why env doesn't work?

**Reply by jsjoeio** [reply]
cc @bpmct any ideas?

**Comment by ozzyst** [comment, marked as answer]
Also, I have noticed. If I set env in the terminal via code server, I will not have access to this.

If I try to get a var from default Kubernetes, I will get it.

```
coder@ide-5784f44f46-l55nk:~$ env
..
USER=coder <- default from K
...
TELEGRAM_TOKEN=1 <- manually from the terminal in Code Server
...
```

Run code

```
	a := os.Getenv("TELEGRAM_TOKEN")  <- a = ""
	b := os.Getenv("USER") <- b = "coder"
	print(a)
	print(b)
```

Output:

```
Starting: /home/coder/go/bin/dlv-dap dap --check-go-version=false --listen=127.0.0.1:39221 --log-dest=3 from /home/coder/go/src/ton-currency/cmd
DAP server listening at: 127.0.0.1:39221
Type 'dlv help' for list of commands.
coder
```

**Reply by ozzyst** [reply]
And I have checked python. The same. I thought maybe the problem was in dlv.

**Reply by jsjoeio** [reply]
Hmm...part of me thinks that setting environment variables manually in the code-server terminal shouldn't work (aka this is working as expected) but maybe I'm missing something. I haven't worked a ton with environment variables in Golang. In JS projects usually we use `dotenv`. I wonder if Go has an equivalent.

**Reply by ozzyst** [reply]
I tried to use [this](https://github.com/kelseyhightower/envconfig), and found [this](https://github.com/joho/godotenv).

One minute, I will check it

**Reply by ozzyst** [reply]
@jsjoeio thanks a lot! It really helps me and works!

**Reply by jsjoeio** [reply]
Hooray!!!

**Comment by bpmct** [comment]
Are you using the helm chart? I was just using and it worked quite nicely for deploying code-server on Kubernetes. If so, you can set env vars by adding this to your chart:
```yaml
extraVars:
  - name: ENV_VAR
  - value: value_here
```

code-server Helm docs: https://coder.com/docs/code-server/latest/helm

**Reply by ozzyst** [reply]
Nope, I use myself manifests.

If I have a lot of projects, will I have a lot of vars in my manifests? I thought about it, but I've decided that it will be not comfortable.
